applications/ShallowWaterApplication/custom_utilities/shallow_water_utilities.cpp

```diff
@@ -214,111 +168,77 @@ void ShallowWaterUtilities::IdentifyWetDomain(ModelPart& rModelPart, Flags WetFl
                 {
                     height_acc += node.FastGetSolutionStepValue(VELOCITY_Z);
                 }
-                it_elem->Set(WetFlag, (height_acc > 0.0));
-            }
-            else if (method == 3) {
-                Geometry<Node<3>>::ShapeFunctionsGradientsType DN_DX(1);
-                geom.ShapeFunctionsIntegrationPointsGradients(DN_DX, GeometryData::GI_GAUSS_1);
-                array_1d<double,3> height_grad = ZeroVector(3);
-                array_1d<double,3> velocity = ZeroVector(3);
-                for (size_t j = 0; j < geom.size(); ++j)
-                {
-                    height_grad[0] += DN_DX[0](j,0) * geom[j].FastGetSolutionStepValue(HEIGHT);
-                    height_grad[1] += DN_DX[0](j,1) * geom[j].FastGetSolutionStepValue(HEIGHT);
-                    velocity += geom[j].FastGetSolutionStepValue(VELOCITY);
-                }
-                velocity /= geom.size();
-
-                double run_up = -inner_prod(height_grad, velocity);
-
-                it_elem->Set(WetFlag, (run_up > 0.0));
+                rElement.Set(WetFlag, (height_acc > 0.0));
             }
         }
-    }
+    });
 }
 
 void ShallowWaterUtilities::ResetDryDomain(ModelPart& rModelPart, double Thickness)
 {
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
-    {
-        auto it_node = rModelPart.NodesBegin() + i;
-        double& height = it_node->FastGetSolutionStepValue(HEIGHT);
+    block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+        double& height = rNode.FastGetSolutionStepValue(HEIGHT);
         if (height < Thickness)
         {
             height = 0.5 * Thickness;
-            it_node->FastGetSolutionStepValue(MOMENTUM) = ZeroVector(3);
+            rNode.FastGetSolutionStepValue(MOMENTUM) = ZeroVector(3);
         }
-    }
+    });
 }
 
 void ShallowWaterUtilities::NormalizeVector(ModelPart& rModelPart, Variable<array_1d<double,3>>& rVariable)
 {
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
-    {
-        auto it_node = rModelPart.NodesBegin() + i;
-        auto& vector = it_node->FastGetSolutionStepValue(rVariable);
+    block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+        auto& vector = rNode.FastGetSolutionStepValue(rVariable);
         const auto modulus = norm_2(vector);
         if (modulus > std::numeric_limits<double>::epsilon())
             vector /= modulus;
-    }
+    });
 }
 
 void ShallowWaterUtilities::SetMinimumValue(ModelPart& rModelPart, const Variable<double>& rVariable, double MinValue)
 {
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
-    {
-        auto& value = (rModelPart.NodesBegin() + i)->FastGetSolutionStepValue(rVariable);
+    block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+        auto& value = rNode.FastGetSolutionStepValue(rVariable);
```

Comment:
```suggestion
        double& value = rNode.FastGetSolutionStepValue(rVariable);
```
For simple return types as `double` it's better not to use `auto`.